src/main/java/htsjdk/samtools/filter/ReadNameFilter.java

```diff
@@ -104,19 +94,8 @@ public boolean filterOut(final SAMRecord record) {
      */
     @Override
     public boolean filterOut(final SAMRecord first, final SAMRecord second) {
-        if (includeReads) {
-            if (readNameFilterSet.contains(first.getReadName()) &&
-                readNameFilterSet.contains(second.getReadName())) {
-                return false;
-            }
-        } else {
-            if (!readNameFilterSet.contains(first.getReadName()) &&
-                !readNameFilterSet.contains(second.getReadName())) {
-                return false;
-            }
-        }
-
-        return true;
+        return (readNameFilterSet.contains(first.getReadName()) &&
+                readNameFilterSet.contains(second.getReadName())) != includeReads;

```

Comment:
The negation of the expression `readNameFilterSet.contains(first.getReadName()) && readNameFilterSet.contains(second.getReadName())` is `! readNameFilterSet.contains(first.getReadName()) || ! readNameFilterSet.contains(second.getReadName())`, which doesn't match the old version from the else clause, which had `(!readNameFilterSet.contains(first.getReadName()) && !readNameFilterSet.contains(second.getReadName()))`